lib/utils.ts

```diff
@@ -59,15 +62,27 @@ export function expandTabs(line: string): string {
     });
 }
 
+function getRegexForTempdir(): RegExp {
+    const tmp = os.tmpdir();
+    return new RegExp(tmp.replaceAll('/', '\\/') + '\\/' + ce_temp_prefix + '[\\w\\d-.]*\\/');
+}
+
+/**
+ * Removes the root dir from the given filepath, so that it will match to the user's filenames used
+ *  note: will keep /app/ if instead of filepath something like '-I/tmp/path' is used
+ * @param filepath
+ * @returns
+ */
 export function maskRootdir(filepath: string): string {
     if (filepath) {
-        // todo: make this compatible with local installations etc
         if (process.platform === 'win32') {
+            // todo: should also use temp_prefix here
             return filepath
                 .replace(/^C:\/Users\/[\w\d-.]*\/AppData\/Local\/Temp\/compiler-explorer-compiler[\w\d-.]*\//, '/app/')
                 .replace(/^\/app\//, '');
         } else {
-            return filepath.replace(/^\/tmp\/compiler-explorer-compiler[\w\d-.]*\//, '/app/').replace(/^\/app\//, '');
+            const re = getRegexForTempdir();
+            return filepath.replace(re, '/app/').replace(/^\/app\//, '');
```

Comment:
replaceAll? And does this replace all tmpdirs with `/app/` and then immediately replace `/app/` with '' ?

Reply:
I put in a description for how it behaves for a reason ;) But it's definitely not a great nor perfect solution

It's done in steps in case for some reason the passed in path was already transformed to /app/...